lib/src/extended_network_image_utils.dart

```diff
@@ -1,105 +1,42 @@
 import 'dart:async';
-import 'dart:convert';
-import 'dart:io';
 import 'dart:typed_data';
-import 'package:crypto/crypto.dart';
-import 'package:flutter/material.dart';
-import 'package:path/path.dart';
-import 'package:path_provider/path_provider.dart';
+import 'package:flutter/painting.dart';
 
+import 'extended_image_utils.dart';
 import 'extended_network_image_provider.dart';
-
-const String cacheImageFolderName = 'cacheimage';
-
-String keyToMd5(String key) => md5.convert(utf8.encode(key)).toString();
+import 'io/extended_network_image_utils.dart'
+    if (dart.library.html) '_extended_network_image_utils_web.dart' as utils;
 
 /// clear the disk cache directory then return if it succeed.
 ///  <param name="duration">timespan to compute whether file has expired or not</param>
 Future<bool> clearDiskCachedImages({Duration duration}) async {
-  try {
-    final Directory cacheImagesDirectory = Directory(
-        join((await getTemporaryDirectory()).path, cacheImageFolderName));
-    if (cacheImagesDirectory.existsSync()) {
-      if (duration == null) {
-        cacheImagesDirectory.deleteSync(recursive: true);
-      } else {
-        final DateTime now = DateTime.now();
-        for (final FileSystemEntity file in cacheImagesDirectory.listSync()) {
-          final FileStat fs = file.statSync();
-          if (now.subtract(duration).isAfter(fs.changed)) {
-            //print("remove expired cached image");
-            file.deleteSync(recursive: true);
-          }
-        }
-      }
-    }
-  } catch (_) {
-    return false;
-  }
-  return true;
+  return utils.clearDiskCachedImages(duration: duration);
 }
 
 /// clear the disk cache image then return if it succeed.
 ///  <param name="url">clear specific one</param>
 Future<bool> clearDiskCachedImage(String url) async {
-  try {
-    final File file = await getCachedImageFile(url);
-    if (file != null) {
-      await file.delete(recursive: true);
-    }
-  } catch (_) {
-    return false;
-  }
-  return true;
+  return utils.clearDiskCachedImage(url);
 }
 
 ///get the local file of the cached image
+Future<String> getCachedImageFilePath(String url) async {
+  return utils.getCachedImageFilePath(url);
+}
```

Comment:
don't make breaking change, may be add new method is better

Reply:
I have considered all possible solutions in this case, and unfortunately, this is the only solution that can ensure compatibility

So that users do not face the problem, I suggest:
1. publish it package with minor version increment 0.4.0 (this will prevent the pub from automatically updating the package and breaking compatibility)
2. publish extended_image package also with minor version increment 1.6.0
3. Warn users in changelog.md about public api change

to update the package to the new version, users change it manually in pubspec, if problems arise, then first of all visit changelog at pub.dev this is normal practice, it is followed by flutter team

also i have experience with same api changes: https://pub.dev/publishers/serge.software/packages